atomic_reactor/util.py

```diff
@@ -719,24 +734,27 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
         media_type = get_manifest_media_type(version)
         headers = {'Accept': media_type}
 
-        for attempts in range(10):
-            response = query_registry(
-                image, registry, digest=None,
-                insecure=insecure, dockercfg_path=dockercfg_path,
-                version=version)
-            if response.status_code == 404:
-                sleep(1)
-            else:
-                break
+        response = query_registry(
+            image, registry, digest=None,
+            insecure=insecure, dockercfg_path=dockercfg_path,
+            version=version)
 
         # If the registry has a v2 manifest that can't be converted into a v1
         # manifest, the registry fails with status=400, and a error code of
-        # MANIFEST_INVALID.
-        if version == 'v1' and response.status_code == 400:
+        # MANIFEST_INVALID. Note that if the registry has v2 manifest and
+        # you ask for an OCI manifest, the registry will try to convert the
+        # v2 manifest into a v1 manifest as the default type, so the same
+        # thing occurs.
+        if version != 'v2' and response.status_code == 400:
             logger.warning('Unable to fetch digest for %s, got error %s',
                            media_type, response.status_code)
             continue
 
+        # Returned if the manifest could not be retrieved for the given
+        # media type
+        if response.status_code == 404 or response.status_code == 406:
```

Comment:
Sure - your call. Once you get beyond 200/404 the numeric status codes are arguably less self-documenting ;-)